lib/matplotlib/__init__.py

```diff
@@ -225,7 +225,7 @@ def _is_writable_dir(p):
             t.write(b'1')
         finally:
             t.close()
-    except OSError:
+    except:
```

Comment:
I thought the restriction that you could only raise `Exception` objects was new-ish and we support back to 2.6 (but I couldn't figure out quickly when that got changed).